### PR TITLE
GUI: Allow texture packs to customize the progress bar

### PR DESCRIPTION
**Pull request by kilbith**
Title says it all.

### CODE REVIEW

File: src/drawscene.cpp [hunk 1]
```diff
@@ -592,30 +598,53 @@ void draw_load_screen(const std::wstring &text, IrrlichtDevice* device,
...
+			progress_img = driver->getTexture(
+				(gamepath + "/base/pack/progress_bar.png").c_str());
+			progress_img_bg = driver->getTexture(
+				(gamepath + "/base/pack/progress_bar_bg.png").c_str());
+		}
+
+		if (progress_img && progress_img_bg) {
+			const core::dimension2d<u32> &img_size = progress_img_bg->getSize();
+			u32 imgW = MYMIN(MYMAX(img_size.Width, 200), 600);
+			u32 imgH = MYMIN(MYMAX(img_size.Height, 24), 72);
```
Comment (SmallJoker) [inline src/drawscene.cpp]: We have the macro `rangelim(d, min, max)`

**Comment by SmallJoker** [comment]
Please also document these file names in the file `doc/texture_packs.txt`.

File: src/drawscene.cpp [hunk 2]
```diff
@@ -32,6 +32,12 @@ typedef enum {
 	EYECOUNT = 2
 } paralax_sign;
 
+inline bool file_exists(const std::string &name)
```
Comment (SmallJoker) [inline src/drawscene.cpp]: This function belongs to `filesys.h`.

File: src/drawscene.cpp [hunk 3]
```diff
@@ -592,30 +598,53 @@ void draw_load_screen(const std::wstring &text, IrrlichtDevice* device,
...
+		if (file_exists(tp_progress_bar) && file_exists(tp_progress_bar_bg)) {
+			progress_img = driver->getTexture((tp_progress_bar).c_str());
+			progress_img_bg = driver->getTexture((tp_progress_bar_bg).c_str());
+		} else {
+			std::string gamepath = fs::RemoveRelativePathComponents(
+				porting::path_share + DIR_DELIM + "textures");
+			progress_img = driver->getTexture(
+				(gamepath + "/base/pack/progress_bar.png").c_str());
+			progress_img_bg = driver->getTexture(
+				(gamepath + "/base/pack/progress_bar_bg.png").c_str());
```
Comment (SmallJoker) [inline src/drawscene.cpp]: This is a form of code duplication. Just update the two strings `tp_progress_bar*` and get the textures afterwards.

**Comment by tobyplowy** [comment]
+1

**Comment by MarkuBu** [comment]
Is this already merged? Because since the update today the progress bar when starting a game has changed. Looks more comicy now

**Comment by sfan5** [comment]
The changed progress bar is already merged, however it can't be customized by texture packs yet (which is what this PR is about).

**Comment by kilbith** [comment]
Updated, thanks for reviewing.

File: src/drawscene.cpp [hunk 4]
```diff
@@ -592,30 +592,51 @@ void draw_load_screen(const std::wstring &text, IrrlichtDevice* device,
...
-			core::rect<s32>(0, 0, img_size.Width, img_size.Height),
-			0, 0, true);
-
-		draw2DImageFilterScaled(
-			driver, progress_img,
-			core::rect<s32>(img_pos.X, img_pos.Y,
-					img_pos.X + (percent * imgW) / 100, img_pos.Y + imgH),
-			core::rect<s32>(0, 0, ((percent * img_size.Width) / 100), img_size.Height),
-			0, 0, true);
+		std::string texture_path = g_settings->get("texture_path");
```
Comment (nerzhul) [inline src/drawscene.cpp]: const std::string &

**Review by nerzhul — APPROVED** [review]
except one copy to remove okay for me

File: src/drawscene.cpp [hunk 5]
```diff
@@ -592,30 +592,51 @@ void draw_load_screen(const std::wstring &text, IrrlichtDevice* device,
...
+			std::string gamepath = fs::RemoveRelativePathComponents(
+				porting::path_share + DIR_DELIM + "textures");
+			tp_progress_bar = gamepath + "/base/pack/progress_bar.png";
+			tp_progress_bar_bg = gamepath + "/base/pack/progress_bar_bg.png";
+		}
+
+		video::ITexture *progress_img =
+			driver->getTexture((tp_progress_bar).c_str());
+		video::ITexture *progress_img_bg =
+			driver->getTexture((tp_progress_bar_bg).c_str());
```
Comment (SmallJoker) [inline src/drawscene.cpp]: Parentheses around `tp_progress_bar*` are redundant

**Comment by Zeno-** [comment]
:+1:

**Review by SmallJoker — APPROVED** [review]

**Pull request merged**